### PR TITLE
Add Person, Student, and Teacher classes

### PR DESCRIPTION
**Pull request by shahira-sadat**
In this PR I 👇🏻:  

Created class Person with the following:

- Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? returns true if a person is of age or if they have permission from parents.

Created class Student with the following:

- Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".

Created class Teacher with the following:

- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.

Each class is a separate file.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+    @classroom = classroom
+    super(
+      age: age,
+      name: name,
+      parent_permission: parent_permission
+    )
```
Comment (ShoiraTa) [inline student.rb]: - [x] When calling super, you don't have to assign variables again, only:
` super(age, name, parent_permission)`
should be enough 👍🏻

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
+    @specialization = specialization
+    super(
+      age: age,
+      name: name,
+      parent_permission: parent_permission
+    )
+  end
```
Comment (ShoiraTa) [inline teacher.rb]: - [x] Same here, please don't assign variables again. just write 
` super(age, name, parent_permission)`

- [x] OPTIONAL: Usually, in ruby apps, `super()` is called right after def `initialize()` before other variable assignments 😉

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+puts 'Enter Teacher Specialization'
+new_specialization = gets.chomp
+puts 'Enter Teacher Age'
+new_age = gets.chomp.to_i
+
+teach = Teacher.new(new_age, new_specialization, new_name)
+puts("Teacher Id: #{teach.id}")
+puts("Teacher Name: #{teach.name}")
+puts("Teacher Specialization: #{teach.specialization}")
+puts("Teacher Age: #{teach.age}")
```
Comment (ShoiraTa) [inline teacher.rb]: - [x] OPTIONAL: To keep codebase clean in main branch, please make it a habit to push only clean code. I strongly suggest to delete all `puts` 👍🏻

File: .github/workflow/linters.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+name: Linters
+
+on: pull_request
+
```
Comment (ShoiraTa) [inline .github/workflow/linters.yml]: - [x] Even though you added linters.yml file, tests are not running on github repository. Please go through [linters installation steps for ruby](https://github.com/microverseinc/linters-config/tree/master/ruby) and make sure that tests are running on repository. Currently there are 0 tests:
Screenshot 2022-09-27 at 11 09 02

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @shahira-sadat ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Great commit history! ✔️ 
- `Person`, `Student`, and `Teacher` classes with relevant methods are present✔️ 
-  PRoject is setup correctly ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username [@shoirata].


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @shahira-sadat,

✔️ Good documentation

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**